libraries/stdlib/src/kotlin/text/Strings.kt

```diff
@@ -1183,8 +1191,53 @@ public fun CharSequence.splitToSequence(vararg delimiters: Char, ignoreCase: Boo
  * @param ignoreCase `true` to ignore character case when matching a delimiter. By default `false`.
  * @param limit The maximum number of substrings to return.
  */
-public fun CharSequence.split(vararg delimiters: Char, ignoreCase: Boolean = false, limit: Int = 0): List<String> =
-        rangesDelimitedBy(delimiters, ignoreCase = ignoreCase, limit = limit).asIterable().map { substring(it) }
+public fun CharSequence.split(vararg delimiters: Char, ignoreCase: Boolean = false, limit: Int = 0): List<String> {
+    if (delimiters.size == 1) {
+        return split(delimiters[0].toString(), ignoreCase, limit)
+    }
+
+    return rangesDelimitedBy(delimiters, ignoreCase = ignoreCase, limit = limit).asIterable().map { substring(it) }
+}
+
+/**
+ * Splits this char sequence to a list of strings around occurrences of the specified [delimiter].
+ * This is specialized version of split which receives single non-empty delimiter and offers better performance
+ *
+ * @param delimiter String used as delimiter
+ * @param ignoreCase `true` to ignore character case when matching a delimiter. By default `false`.
+ * @param limit The maximum number of substrings to return.
+ */
+private fun CharSequence.split(delimiter: String, ignoreCase: Boolean, limit: Int): List<String> {
+    var currentOffset = 0
+    var nextIndex = indexOf(delimiter, currentOffset, ignoreCase)
+    val isLimited = limit > 0
+    val result = ArrayList<String>()
```

Comment:
Agree, please do, but only for `isLimited` case.
Better to use `minOf`, it's available without import.

Reply:
Why not in both cases?
10 (not 16 actually) is default array list size in both openjdk and android.

`result = ArrayList<String>(minOf(limit, 10))` looks simpler than `result = if (isLimited) ... else ...`

Reply:
`minOf(limit, 10)` is 0 when the limit is 0. Is the capacity intended to be so?
Actually it would be better to use `limit.coerceAtMost(10)` here to communicate the intent more clearly.

Reply:
Yes. "0" indicates that default size should be used

Reply:
As far as I can see in the sources, `ArrayList()` and `ArrayList(0)` constructors initialize the list state differently, the latter grows its capacity more often, so it may result in more reallocations.

Reply:
Yes, you're right